### DISCUSSION TITLE
Storage pools bug?

### DISCUSSION THREAD

**Discussion by Verhoeckx** [cockpit-project/cockpit-machines#291] [General]
I have the impression that the size of the different storage pools aren't calculated correctly:

![Screenshot from 2021-07-29 16-37-55](https://user-images.githubusercontent.com/64807887/127511862-6400274b-c4f5-47c3-ade7-d7ebbca53686.png)
![Screenshot from 2021-07-29 16-38-36](https://user-images.githubusercontent.com/64807887/127511865-0aa2cb59-919a-49fc-a34e-1dba2c9a6d43.png)

Both storage pools display a size of 15.07 GB.

However, the pool `images` only contains 9.43 GB (the RHEL ISO file) and the pool `Main storage` is only  used for 1.85 GB (at the moment.

Shouldn't both pools just display what they contain? Possibly a bug?

**Comment by KKoukiou** [comment]
Hey @Verhoeckx, thanks for submitting the bug report!
The information we show in the UI is from libvirt aka on my system:
```
$ sudo virsh pool-info Downloads
Name:           Downloads
UUID:           7d6de773-9610-4556-bc18-725d2a8f8a3a
State:          running
Persistent:     yes
Autostart:      yes
Capacity:       179.37 GiB
Allocation:     132.81 GiB
Available:      46.55 GiB
```

So on the title of the slider you see: Allocation / Capacity.

The fact that you noticed, that the sum of the volumes sizes don't add up to the allocation of the pool themselves is totally normal for directory pools. Image that your `Images` pool, which is probably something on your root partition mounted on /, like /var/lib/libvirt/images has the whole space on that partition to expand, it's not just that directory contents that are displayed in the Slider.

Same for the Main storage pool, if it's on your home partition for example, which has 2216GiB capacity, and the allocation is the free space on that partition.

**Reply by Verhoeckx** [reply]
Ah, this is confusing: 
The command `sudo virsh pool-info [name storage pool]` _doesn't_ show the capacity / allocation / availability of the storage pool, it does show the capacity / allocation / availability of the partition on which they live, in both cases the root partition.

<strike>So using it for the capacity / allocation / availability of the root filesystem it's fine but it just simple doesn't work for the storage pools themself.</strike> Maybe there is another command that can be used for this? Many Linux commands do not return what you suspect.

As a user I expect the slider of the storage pools give the _exact_ availability and capacity that they actually use. Otherwise the information becomes useless.

**Reply by KKoukiou** [reply]
I understand your confusion/frustration, but in cockpit we don't do that kind of guessing, but rather use other system APIs to get the output we present in the UI.

I think it's a conversation to be done with libvirt, which is the library that provides that 'virsh' command from above, to decide if the output of pool-info makes sense for directory pools.

**Reply by Verhoeckx** [reply]
I was thinking the exact same thing.

I guess there are two types of storage pools: one who have there own partition, and one who are just a directory. The command works fine for the first type but doesn't work for the second type.
